Kernel/Syscalls/sched.cpp

```diff
@@ -17,48 +19,69 @@ ErrorOr<FlatPtr> Process::sys$yield()
     return 0;
 }
 
-ErrorOr<FlatPtr> Process::sys$sched_setparam(int pid, Userspace<const struct sched_param*> user_param)
+ErrorOr<NonnullRefPtr<Thread>> Process::get_thread_from_pid_or_tid(pid_t pid_or_tid, Syscall::SchedulerParametersMode mode)
+{
+    VERIFY(g_scheduler_lock.is_locked_by_current_processor());
+    Thread* peer;
+    if (mode == Syscall::SchedulerParametersMode::Thread) {
+        peer = Thread::current();
+        if (pid_or_tid != 0)
+            peer = Thread::from_tid(pid_or_tid);
+
+        // Syscall::SchedulerParametersMode::Process
+    } else {
+        auto* searched_process = this;
+        if (pid_or_tid != 0)
+            searched_process = Process::from_pid(pid_or_tid);
+
+        if (searched_process == nullptr)
+            return ESRCH;
+        auto pid = searched_process->pid().value();
+        // Main thread has tid == pid
+        this->thread_list().for_each([&](auto& thread) {
+            if (thread.tid().value() == pid)
+                peer = &thread;
+        });
+    }
```

Comment:
```suggestion
    } else {
        VERIFY_NOT_REACHED();
    }
```

This is in conjunction with the previous comment, as large suggestion blocks make the suggestion unreadable.

(Yes, this is what we discussed earlier already, but I believe I didn't bring my point across clearly enough.)